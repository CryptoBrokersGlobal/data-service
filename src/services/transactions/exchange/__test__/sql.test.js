const sql = require('../sql');
const filters = {
  timeStart: 'timeStart',
  timeEnd: 'timeEnd',
  amountAsset: 'amountAsset',
  priceAsset: 'priceAsset',
  matcher: 'matcher',
  sender: 'sender',
  sort: 'sort',
  limit: 1,
};

describe('Sql builder', () => {
  describe('search', () => {
    it('covers case with all filters (without after)', () => {
      expect(sql.search(filters)).toMatchSnapshot();
    });
    it('covers case with all filters with after', () => {
      expect(
        sql.search({
          ...filters,
          after: {
            timestamp: 'timestamp',
            id: 'id',
            sortDirection: 'sortDirection',
          },
        })
      ).toMatchSnapshot();
    });
  });

  describe('get', () => {
    it('works', () => {
      expect(sql.get('id')).toMatchSnapshot();
    });
  });

  describe('mget', () => {
    it('works', () => {
      expect(sql.mget(['id1', 'id2'])).toMatchSnapshot();
    });
  });
});
