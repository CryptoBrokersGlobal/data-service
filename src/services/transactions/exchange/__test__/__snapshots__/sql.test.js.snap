// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sql builder get works 1`] = `"select \\"tx_id\\", \\"tx_time_stamp\\", \\"tx_height\\", \\"tx_signature\\", \\"tx_sender\\", \\"tx_sender_public_key\\", \\"tx_amount_asset\\", \\"tx_price_asset\\", \\"tx_fee\\", \\"tx_sell_matcher_fee\\", \\"tx_buy_matcher_fee\\", \\"o1_id\\", \\"o1_time_stamp\\", \\"o1_expiration\\", \\"o1_signature\\", \\"o1_sender\\", \\"o1_sender_public_key\\", \\"o1_type\\", \\"o1_matcher_fee\\", \\"o2_id\\", \\"o2_time_stamp\\", \\"o2_expiration\\", \\"o2_signature\\", \\"o2_sender\\", \\"o2_sender_public_key\\", \\"o2_type\\", \\"o2_matcher_fee\\", t.tx_price * 10^(-8 - p_dec.decimals + a_dec.decimals) as \\"tx_price\\", t.tx_amount * 10^(-a_dec.decimals) as \\"tx_amount\\", t.o1_price * 10^(-8 - p_dec.decimals + a_dec.decimals) as \\"o1_price\\", t.o1_amount * 10^(-a_dec.decimals) as \\"o1_amount\\", t.o2_price * 10^(-8 - p_dec.decimals + a_dec.decimals) as \\"o2_price\\", t.o2_amount * 10^(-a_dec.decimals) as \\"o2_amount\\" from (select \\"t\\".\\"id\\" as \\"tx_id\\", \\"t\\".\\"time_stamp\\" as \\"tx_time_stamp\\", \\"t\\".\\"height\\" as \\"tx_height\\", \\"t\\".\\"signature\\" as \\"tx_signature\\", \\"t\\".\\"sender\\" as \\"tx_sender\\", \\"t\\".\\"sender_public_key\\" as \\"tx_sender_public_key\\", \\"t\\".\\"amount_asset\\" as \\"tx_amount_asset\\", \\"t\\".\\"price_asset\\" as \\"tx_price_asset\\", \\"t\\".\\"price\\" as \\"tx_price\\", \\"t\\".\\"amount\\" as \\"tx_amount\\", t.fee * 10^(-8) as \\"tx_fee\\", t.sell_matcher_fee * 10^(-8) as \\"tx_sell_matcher_fee\\", t.buy_matcher_fee * 10^(-8) as \\"tx_buy_matcher_fee\\", \\"o1_id\\" as \\"o1_id\\", \\"o1_time_stamp\\" as \\"o1_time_stamp\\", \\"o1_expiration\\" as \\"o1_expiration\\", \\"o1_signature\\" as \\"o1_signature\\", \\"o1_sender\\" as \\"o1_sender\\", \\"o1_sender_public_key\\" as \\"o1_sender_public_key\\", \\"o1_type\\" as \\"o1_type\\", \\"o1_price\\" as \\"o1_price\\", \\"o1_amount\\" as \\"o1_amount\\", \\"o1_matcher_fee\\" as \\"o1_matcher_fee\\", \\"o2_id\\" as \\"o2_id\\", \\"o2_time_stamp\\" as \\"o2_time_stamp\\", \\"o2_expiration\\" as \\"o2_expiration\\", \\"o2_signature\\" as \\"o2_signature\\", \\"o2_sender\\" as \\"o2_sender\\", \\"o2_sender_public_key\\" as \\"o2_sender_public_key\\", \\"o2_type\\" as \\"o2_type\\", \\"o2_price\\" as \\"o2_price\\", \\"o2_amount\\" as \\"o2_amount\\", \\"o2_matcher_fee\\" as \\"o2_matcher_fee\\" from (select \\"t\\".\\"id\\", \\"t\\".\\"time_stamp\\", \\"t\\".\\"height\\", \\"t\\".\\"signature\\", \\"t\\".\\"sender\\", \\"t\\".\\"sender_public_key\\", \\"t\\".\\"amount_asset\\", \\"t\\".\\"price_asset\\", \\"t\\".\\"price\\", \\"t\\".\\"amount\\", \\"t\\".\\"fee\\", \\"t\\".\\"sell_matcher_fee\\", \\"t\\".\\"buy_matcher_fee\\", \\"t\\".\\"order1\\", \\"t\\".\\"order2\\", \\"o1\\".\\"id\\" as \\"o1_id\\", \\"o1\\".\\"time_stamp\\" as \\"o1_time_stamp\\", \\"o1\\".\\"expiration\\" as \\"o1_expiration\\", \\"o1\\".\\"signature\\" as \\"o1_signature\\", \\"o1\\".\\"sender\\" as \\"o1_sender\\", \\"o1\\".\\"sender_public_key\\" as \\"o1_sender_public_key\\", \\"o1\\".\\"order_type\\" as \\"o1_type\\", \\"o1\\".\\"price\\" as \\"o1_price\\", \\"o1\\".\\"amount\\" as \\"o1_amount\\", o1.matcher_fee * 10^(-8) as \\"o1_matcher_fee\\", \\"o2\\".\\"id\\" as \\"o2_id\\", \\"o2\\".\\"time_stamp\\" as \\"o2_time_stamp\\", \\"o2\\".\\"expiration\\" as \\"o2_expiration\\", \\"o2\\".\\"signature\\" as \\"o2_signature\\", \\"o2\\".\\"sender\\" as \\"o2_sender\\", \\"o2\\".\\"sender_public_key\\" as \\"o2_sender_public_key\\", \\"o2\\".\\"order_type\\" as \\"o2_type\\", \\"o2\\".\\"price\\" as \\"o2_price\\", \\"o2\\".\\"amount\\" as \\"o2_amount\\", o2.matcher_fee * 10^(-8) as \\"o2_matcher_fee\\" from (select \\"t\\".\\"id\\", \\"t\\".\\"time_stamp\\", \\"t\\".\\"height\\", \\"t\\".\\"signature\\", \\"t\\".\\"sender\\", \\"t\\".\\"sender_public_key\\", \\"t\\".\\"amount_asset\\", \\"t\\".\\"price_asset\\", \\"t\\".\\"price\\", \\"t\\".\\"amount\\", \\"t\\".\\"fee\\", \\"t\\".\\"sell_matcher_fee\\", \\"t\\".\\"buy_matcher_fee\\", \\"t\\".\\"order1\\", \\"t\\".\\"order2\\" from \\"txs_7\\" as \\"t\\" where \\"t\\".\\"id\\" = 'id') as \\"t\\" inner join \\"orders\\" as \\"o1\\" on \\"t\\".\\"order1\\" = \\"o1\\".\\"id\\" inner join \\"orders\\" as \\"o2\\" on \\"t\\".\\"order2\\" = \\"o2\\".\\"id\\") as \\"t\\") as \\"t\\" inner join \\"asset_decimals\\" as \\"a_dec\\" on \\"t\\".\\"tx_amount_asset\\" = \\"a_dec\\".\\"asset_id\\" inner join \\"asset_decimals\\" as \\"p_dec\\" on \\"t\\".\\"tx_price_asset\\" = \\"p_dec\\".\\"asset_id\\""`;

exports[`Sql builder mget works 1`] = `"select \\"tx_id\\", \\"tx_time_stamp\\", \\"tx_height\\", \\"tx_signature\\", \\"tx_sender\\", \\"tx_sender_public_key\\", \\"tx_amount_asset\\", \\"tx_price_asset\\", \\"tx_fee\\", \\"tx_sell_matcher_fee\\", \\"tx_buy_matcher_fee\\", \\"o1_id\\", \\"o1_time_stamp\\", \\"o1_expiration\\", \\"o1_signature\\", \\"o1_sender\\", \\"o1_sender_public_key\\", \\"o1_type\\", \\"o1_matcher_fee\\", \\"o2_id\\", \\"o2_time_stamp\\", \\"o2_expiration\\", \\"o2_signature\\", \\"o2_sender\\", \\"o2_sender_public_key\\", \\"o2_type\\", \\"o2_matcher_fee\\", t.tx_price * 10^(-8 - p_dec.decimals + a_dec.decimals) as \\"tx_price\\", t.tx_amount * 10^(-a_dec.decimals) as \\"tx_amount\\", t.o1_price * 10^(-8 - p_dec.decimals + a_dec.decimals) as \\"o1_price\\", t.o1_amount * 10^(-a_dec.decimals) as \\"o1_amount\\", t.o2_price * 10^(-8 - p_dec.decimals + a_dec.decimals) as \\"o2_price\\", t.o2_amount * 10^(-a_dec.decimals) as \\"o2_amount\\" from (select \\"t\\".\\"id\\" as \\"tx_id\\", \\"t\\".\\"time_stamp\\" as \\"tx_time_stamp\\", \\"t\\".\\"height\\" as \\"tx_height\\", \\"t\\".\\"signature\\" as \\"tx_signature\\", \\"t\\".\\"sender\\" as \\"tx_sender\\", \\"t\\".\\"sender_public_key\\" as \\"tx_sender_public_key\\", \\"t\\".\\"amount_asset\\" as \\"tx_amount_asset\\", \\"t\\".\\"price_asset\\" as \\"tx_price_asset\\", \\"t\\".\\"price\\" as \\"tx_price\\", \\"t\\".\\"amount\\" as \\"tx_amount\\", t.fee * 10^(-8) as \\"tx_fee\\", t.sell_matcher_fee * 10^(-8) as \\"tx_sell_matcher_fee\\", t.buy_matcher_fee * 10^(-8) as \\"tx_buy_matcher_fee\\", \\"o1_id\\" as \\"o1_id\\", \\"o1_time_stamp\\" as \\"o1_time_stamp\\", \\"o1_expiration\\" as \\"o1_expiration\\", \\"o1_signature\\" as \\"o1_signature\\", \\"o1_sender\\" as \\"o1_sender\\", \\"o1_sender_public_key\\" as \\"o1_sender_public_key\\", \\"o1_type\\" as \\"o1_type\\", \\"o1_price\\" as \\"o1_price\\", \\"o1_amount\\" as \\"o1_amount\\", \\"o1_matcher_fee\\" as \\"o1_matcher_fee\\", \\"o2_id\\" as \\"o2_id\\", \\"o2_time_stamp\\" as \\"o2_time_stamp\\", \\"o2_expiration\\" as \\"o2_expiration\\", \\"o2_signature\\" as \\"o2_signature\\", \\"o2_sender\\" as \\"o2_sender\\", \\"o2_sender_public_key\\" as \\"o2_sender_public_key\\", \\"o2_type\\" as \\"o2_type\\", \\"o2_price\\" as \\"o2_price\\", \\"o2_amount\\" as \\"o2_amount\\", \\"o2_matcher_fee\\" as \\"o2_matcher_fee\\" from (select \\"t\\".\\"id\\", \\"t\\".\\"time_stamp\\", \\"t\\".\\"height\\", \\"t\\".\\"signature\\", \\"t\\".\\"sender\\", \\"t\\".\\"sender_public_key\\", \\"t\\".\\"amount_asset\\", \\"t\\".\\"price_asset\\", \\"t\\".\\"price\\", \\"t\\".\\"amount\\", \\"t\\".\\"fee\\", \\"t\\".\\"sell_matcher_fee\\", \\"t\\".\\"buy_matcher_fee\\", \\"t\\".\\"order1\\", \\"t\\".\\"order2\\", \\"o1\\".\\"id\\" as \\"o1_id\\", \\"o1\\".\\"time_stamp\\" as \\"o1_time_stamp\\", \\"o1\\".\\"expiration\\" as \\"o1_expiration\\", \\"o1\\".\\"signature\\" as \\"o1_signature\\", \\"o1\\".\\"sender\\" as \\"o1_sender\\", \\"o1\\".\\"sender_public_key\\" as \\"o1_sender_public_key\\", \\"o1\\".\\"order_type\\" as \\"o1_type\\", \\"o1\\".\\"price\\" as \\"o1_price\\", \\"o1\\".\\"amount\\" as \\"o1_amount\\", o1.matcher_fee * 10^(-8) as \\"o1_matcher_fee\\", \\"o2\\".\\"id\\" as \\"o2_id\\", \\"o2\\".\\"time_stamp\\" as \\"o2_time_stamp\\", \\"o2\\".\\"expiration\\" as \\"o2_expiration\\", \\"o2\\".\\"signature\\" as \\"o2_signature\\", \\"o2\\".\\"sender\\" as \\"o2_sender\\", \\"o2\\".\\"sender_public_key\\" as \\"o2_sender_public_key\\", \\"o2\\".\\"order_type\\" as \\"o2_type\\", \\"o2\\".\\"price\\" as \\"o2_price\\", \\"o2\\".\\"amount\\" as \\"o2_amount\\", o2.matcher_fee * 10^(-8) as \\"o2_matcher_fee\\" from (select \\"t\\".\\"id\\", \\"t\\".\\"time_stamp\\", \\"t\\".\\"height\\", \\"t\\".\\"signature\\", \\"t\\".\\"sender\\", \\"t\\".\\"sender_public_key\\", \\"t\\".\\"amount_asset\\", \\"t\\".\\"price_asset\\", \\"t\\".\\"price\\", \\"t\\".\\"amount\\", \\"t\\".\\"fee\\", \\"t\\".\\"sell_matcher_fee\\", \\"t\\".\\"buy_matcher_fee\\", \\"t\\".\\"order1\\", \\"t\\".\\"order2\\" from \\"txs_7\\" as \\"t\\" where \\"t\\".\\"id\\" in ('id1', 'id2')) as \\"t\\" inner join \\"orders\\" as \\"o1\\" on \\"t\\".\\"order1\\" = \\"o1\\".\\"id\\" inner join \\"orders\\" as \\"o2\\" on \\"t\\".\\"order2\\" = \\"o2\\".\\"id\\") as \\"t\\") as \\"t\\" inner join \\"asset_decimals\\" as \\"a_dec\\" on \\"t\\".\\"tx_amount_asset\\" = \\"a_dec\\".\\"asset_id\\" inner join \\"asset_decimals\\" as \\"p_dec\\" on \\"t\\".\\"tx_price_asset\\" = \\"p_dec\\".\\"asset_id\\""`;

exports[`Sql builder search covers case with all filters (without after) 1`] = `"select \\"tx_id\\", \\"tx_time_stamp\\", \\"tx_height\\", \\"tx_signature\\", \\"tx_sender\\", \\"tx_sender_public_key\\", \\"tx_amount_asset\\", \\"tx_price_asset\\", \\"tx_fee\\", \\"tx_sell_matcher_fee\\", \\"tx_buy_matcher_fee\\", \\"o1_id\\", \\"o1_time_stamp\\", \\"o1_expiration\\", \\"o1_signature\\", \\"o1_sender\\", \\"o1_sender_public_key\\", \\"o1_type\\", \\"o1_matcher_fee\\", \\"o2_id\\", \\"o2_time_stamp\\", \\"o2_expiration\\", \\"o2_signature\\", \\"o2_sender\\", \\"o2_sender_public_key\\", \\"o2_type\\", \\"o2_matcher_fee\\", t.tx_price * 10^(-8 - p_dec.decimals + a_dec.decimals) as \\"tx_price\\", t.tx_amount * 10^(-a_dec.decimals) as \\"tx_amount\\", t.o1_price * 10^(-8 - p_dec.decimals + a_dec.decimals) as \\"o1_price\\", t.o1_amount * 10^(-a_dec.decimals) as \\"o1_amount\\", t.o2_price * 10^(-8 - p_dec.decimals + a_dec.decimals) as \\"o2_price\\", t.o2_amount * 10^(-a_dec.decimals) as \\"o2_amount\\" from (select \\"t\\".\\"id\\" as \\"tx_id\\", \\"t\\".\\"time_stamp\\" as \\"tx_time_stamp\\", \\"t\\".\\"height\\" as \\"tx_height\\", \\"t\\".\\"signature\\" as \\"tx_signature\\", \\"t\\".\\"sender\\" as \\"tx_sender\\", \\"t\\".\\"sender_public_key\\" as \\"tx_sender_public_key\\", \\"t\\".\\"amount_asset\\" as \\"tx_amount_asset\\", \\"t\\".\\"price_asset\\" as \\"tx_price_asset\\", \\"t\\".\\"price\\" as \\"tx_price\\", \\"t\\".\\"amount\\" as \\"tx_amount\\", t.fee * 10^(-8) as \\"tx_fee\\", t.sell_matcher_fee * 10^(-8) as \\"tx_sell_matcher_fee\\", t.buy_matcher_fee * 10^(-8) as \\"tx_buy_matcher_fee\\", \\"o1_id\\" as \\"o1_id\\", \\"o1_time_stamp\\" as \\"o1_time_stamp\\", \\"o1_expiration\\" as \\"o1_expiration\\", \\"o1_signature\\" as \\"o1_signature\\", \\"o1_sender\\" as \\"o1_sender\\", \\"o1_sender_public_key\\" as \\"o1_sender_public_key\\", \\"o1_type\\" as \\"o1_type\\", \\"o1_price\\" as \\"o1_price\\", \\"o1_amount\\" as \\"o1_amount\\", \\"o1_matcher_fee\\" as \\"o1_matcher_fee\\", \\"o2_id\\" as \\"o2_id\\", \\"o2_time_stamp\\" as \\"o2_time_stamp\\", \\"o2_expiration\\" as \\"o2_expiration\\", \\"o2_signature\\" as \\"o2_signature\\", \\"o2_sender\\" as \\"o2_sender\\", \\"o2_sender_public_key\\" as \\"o2_sender_public_key\\", \\"o2_type\\" as \\"o2_type\\", \\"o2_price\\" as \\"o2_price\\", \\"o2_amount\\" as \\"o2_amount\\", \\"o2_matcher_fee\\" as \\"o2_matcher_fee\\" from (select \\"t\\".\\"id\\", \\"t\\".\\"time_stamp\\", \\"t\\".\\"height\\", \\"t\\".\\"signature\\", \\"t\\".\\"sender\\", \\"t\\".\\"sender_public_key\\", \\"t\\".\\"amount_asset\\", \\"t\\".\\"price_asset\\", \\"t\\".\\"price\\", \\"t\\".\\"amount\\", \\"t\\".\\"fee\\", \\"t\\".\\"sell_matcher_fee\\", \\"t\\".\\"buy_matcher_fee\\", \\"t\\".\\"order1\\", \\"t\\".\\"order2\\", \\"o1\\".\\"id\\" as \\"o1_id\\", \\"o1\\".\\"time_stamp\\" as \\"o1_time_stamp\\", \\"o1\\".\\"expiration\\" as \\"o1_expiration\\", \\"o1\\".\\"signature\\" as \\"o1_signature\\", \\"o1\\".\\"sender\\" as \\"o1_sender\\", \\"o1\\".\\"sender_public_key\\" as \\"o1_sender_public_key\\", \\"o1\\".\\"order_type\\" as \\"o1_type\\", \\"o1\\".\\"price\\" as \\"o1_price\\", \\"o1\\".\\"amount\\" as \\"o1_amount\\", o1.matcher_fee * 10^(-8) as \\"o1_matcher_fee\\", \\"o2\\".\\"id\\" as \\"o2_id\\", \\"o2\\".\\"time_stamp\\" as \\"o2_time_stamp\\", \\"o2\\".\\"expiration\\" as \\"o2_expiration\\", \\"o2\\".\\"signature\\" as \\"o2_signature\\", \\"o2\\".\\"sender\\" as \\"o2_sender\\", \\"o2\\".\\"sender_public_key\\" as \\"o2_sender_public_key\\", \\"o2\\".\\"order_type\\" as \\"o2_type\\", \\"o2\\".\\"price\\" as \\"o2_price\\", \\"o2\\".\\"amount\\" as \\"o2_amount\\", o2.matcher_fee * 10^(-8) as \\"o2_matcher_fee\\" from (select \\"t\\".\\"id\\", \\"t\\".\\"time_stamp\\", \\"t\\".\\"height\\", \\"t\\".\\"signature\\", \\"t\\".\\"sender\\", \\"t\\".\\"sender_public_key\\", \\"t\\".\\"amount_asset\\", \\"t\\".\\"price_asset\\", \\"t\\".\\"price\\", \\"t\\".\\"amount\\", \\"t\\".\\"fee\\", \\"t\\".\\"sell_matcher_fee\\", \\"t\\".\\"buy_matcher_fee\\", \\"t\\".\\"order1\\", \\"t\\".\\"order2\\" from \\"txs_7\\" as \\"t\\" inner join \\"orders\\" as \\"o\\" on \\"t\\".\\"order1\\" = \\"o\\".\\"id\\" or \\"t\\".\\"order2\\" = \\"o\\".\\"id\\" where \\"t\\".\\"time_stamp\\" >= 'timeStart' and \\"t\\".\\"time_stamp\\" <= 'timeEnd' and \\"t\\".\\"amount_asset\\" = 'amountAsset' and \\"t\\".\\"price_asset\\" = 'priceAsset' and \\"t\\".\\"sender\\" = 'matcher' and \\"o\\".\\"sender\\" = 'sender' order by \\"t\\".\\"time_stamp\\" asc, \\"t\\".\\"id\\" asc limit 1) as \\"t\\" inner join \\"orders\\" as \\"o1\\" on \\"t\\".\\"order1\\" = \\"o1\\".\\"id\\" inner join \\"orders\\" as \\"o2\\" on \\"t\\".\\"order2\\" = \\"o2\\".\\"id\\") as \\"t\\") as \\"t\\" inner join \\"asset_decimals\\" as \\"a_dec\\" on \\"t\\".\\"tx_amount_asset\\" = \\"a_dec\\".\\"asset_id\\" inner join \\"asset_decimals\\" as \\"p_dec\\" on \\"t\\".\\"tx_price_asset\\" = \\"p_dec\\".\\"asset_id\\" order by \\"tx_time_stamp\\" asc, \\"tx_id\\" asc"`;

exports[`Sql builder search covers case with all filters with after 1`] = `"select \\"tx_id\\", \\"tx_time_stamp\\", \\"tx_height\\", \\"tx_signature\\", \\"tx_sender\\", \\"tx_sender_public_key\\", \\"tx_amount_asset\\", \\"tx_price_asset\\", \\"tx_fee\\", \\"tx_sell_matcher_fee\\", \\"tx_buy_matcher_fee\\", \\"o1_id\\", \\"o1_time_stamp\\", \\"o1_expiration\\", \\"o1_signature\\", \\"o1_sender\\", \\"o1_sender_public_key\\", \\"o1_type\\", \\"o1_matcher_fee\\", \\"o2_id\\", \\"o2_time_stamp\\", \\"o2_expiration\\", \\"o2_signature\\", \\"o2_sender\\", \\"o2_sender_public_key\\", \\"o2_type\\", \\"o2_matcher_fee\\", t.tx_price * 10^(-8 - p_dec.decimals + a_dec.decimals) as \\"tx_price\\", t.tx_amount * 10^(-a_dec.decimals) as \\"tx_amount\\", t.o1_price * 10^(-8 - p_dec.decimals + a_dec.decimals) as \\"o1_price\\", t.o1_amount * 10^(-a_dec.decimals) as \\"o1_amount\\", t.o2_price * 10^(-8 - p_dec.decimals + a_dec.decimals) as \\"o2_price\\", t.o2_amount * 10^(-a_dec.decimals) as \\"o2_amount\\" from (select \\"t\\".\\"id\\" as \\"tx_id\\", \\"t\\".\\"time_stamp\\" as \\"tx_time_stamp\\", \\"t\\".\\"height\\" as \\"tx_height\\", \\"t\\".\\"signature\\" as \\"tx_signature\\", \\"t\\".\\"sender\\" as \\"tx_sender\\", \\"t\\".\\"sender_public_key\\" as \\"tx_sender_public_key\\", \\"t\\".\\"amount_asset\\" as \\"tx_amount_asset\\", \\"t\\".\\"price_asset\\" as \\"tx_price_asset\\", \\"t\\".\\"price\\" as \\"tx_price\\", \\"t\\".\\"amount\\" as \\"tx_amount\\", t.fee * 10^(-8) as \\"tx_fee\\", t.sell_matcher_fee * 10^(-8) as \\"tx_sell_matcher_fee\\", t.buy_matcher_fee * 10^(-8) as \\"tx_buy_matcher_fee\\", \\"o1_id\\" as \\"o1_id\\", \\"o1_time_stamp\\" as \\"o1_time_stamp\\", \\"o1_expiration\\" as \\"o1_expiration\\", \\"o1_signature\\" as \\"o1_signature\\", \\"o1_sender\\" as \\"o1_sender\\", \\"o1_sender_public_key\\" as \\"o1_sender_public_key\\", \\"o1_type\\" as \\"o1_type\\", \\"o1_price\\" as \\"o1_price\\", \\"o1_amount\\" as \\"o1_amount\\", \\"o1_matcher_fee\\" as \\"o1_matcher_fee\\", \\"o2_id\\" as \\"o2_id\\", \\"o2_time_stamp\\" as \\"o2_time_stamp\\", \\"o2_expiration\\" as \\"o2_expiration\\", \\"o2_signature\\" as \\"o2_signature\\", \\"o2_sender\\" as \\"o2_sender\\", \\"o2_sender_public_key\\" as \\"o2_sender_public_key\\", \\"o2_type\\" as \\"o2_type\\", \\"o2_price\\" as \\"o2_price\\", \\"o2_amount\\" as \\"o2_amount\\", \\"o2_matcher_fee\\" as \\"o2_matcher_fee\\" from (select \\"t\\".\\"id\\", \\"t\\".\\"time_stamp\\", \\"t\\".\\"height\\", \\"t\\".\\"signature\\", \\"t\\".\\"sender\\", \\"t\\".\\"sender_public_key\\", \\"t\\".\\"amount_asset\\", \\"t\\".\\"price_asset\\", \\"t\\".\\"price\\", \\"t\\".\\"amount\\", \\"t\\".\\"fee\\", \\"t\\".\\"sell_matcher_fee\\", \\"t\\".\\"buy_matcher_fee\\", \\"t\\".\\"order1\\", \\"t\\".\\"order2\\", \\"o1\\".\\"id\\" as \\"o1_id\\", \\"o1\\".\\"time_stamp\\" as \\"o1_time_stamp\\", \\"o1\\".\\"expiration\\" as \\"o1_expiration\\", \\"o1\\".\\"signature\\" as \\"o1_signature\\", \\"o1\\".\\"sender\\" as \\"o1_sender\\", \\"o1\\".\\"sender_public_key\\" as \\"o1_sender_public_key\\", \\"o1\\".\\"order_type\\" as \\"o1_type\\", \\"o1\\".\\"price\\" as \\"o1_price\\", \\"o1\\".\\"amount\\" as \\"o1_amount\\", o1.matcher_fee * 10^(-8) as \\"o1_matcher_fee\\", \\"o2\\".\\"id\\" as \\"o2_id\\", \\"o2\\".\\"time_stamp\\" as \\"o2_time_stamp\\", \\"o2\\".\\"expiration\\" as \\"o2_expiration\\", \\"o2\\".\\"signature\\" as \\"o2_signature\\", \\"o2\\".\\"sender\\" as \\"o2_sender\\", \\"o2\\".\\"sender_public_key\\" as \\"o2_sender_public_key\\", \\"o2\\".\\"order_type\\" as \\"o2_type\\", \\"o2\\".\\"price\\" as \\"o2_price\\", \\"o2\\".\\"amount\\" as \\"o2_amount\\", o2.matcher_fee * 10^(-8) as \\"o2_matcher_fee\\" from (select \\"t\\".\\"id\\", \\"t\\".\\"time_stamp\\", \\"t\\".\\"height\\", \\"t\\".\\"signature\\", \\"t\\".\\"sender\\", \\"t\\".\\"sender_public_key\\", \\"t\\".\\"amount_asset\\", \\"t\\".\\"price_asset\\", \\"t\\".\\"price\\", \\"t\\".\\"amount\\", \\"t\\".\\"fee\\", \\"t\\".\\"sell_matcher_fee\\", \\"t\\".\\"buy_matcher_fee\\", \\"t\\".\\"order1\\", \\"t\\".\\"order2\\" from \\"txs_7\\" as \\"t\\" inner join \\"orders\\" as \\"o\\" on \\"t\\".\\"order1\\" = \\"o\\".\\"id\\" or \\"t\\".\\"order2\\" = \\"o\\".\\"id\\" where (t.time_stamp, t.id) > ('timestamp', 'id') and \\"t\\".\\"time_stamp\\" >= 'timeStart' and \\"t\\".\\"time_stamp\\" <= 'timeEnd' and \\"t\\".\\"amount_asset\\" = 'amountAsset' and \\"t\\".\\"price_asset\\" = 'priceAsset' and \\"t\\".\\"sender\\" = 'matcher' and \\"o\\".\\"sender\\" = 'sender' order by \\"t\\".\\"time_stamp\\" asc, \\"t\\".\\"id\\" asc limit 1) as \\"t\\" inner join \\"orders\\" as \\"o1\\" on \\"t\\".\\"order1\\" = \\"o1\\".\\"id\\" inner join \\"orders\\" as \\"o2\\" on \\"t\\".\\"order2\\" = \\"o2\\".\\"id\\") as \\"t\\") as \\"t\\" inner join \\"asset_decimals\\" as \\"a_dec\\" on \\"t\\".\\"tx_amount_asset\\" = \\"a_dec\\".\\"asset_id\\" inner join \\"asset_decimals\\" as \\"p_dec\\" on \\"t\\".\\"tx_price_asset\\" = \\"p_dec\\".\\"asset_id\\" order by \\"tx_time_stamp\\" asc, \\"tx_id\\" asc"`;
